src/expr/src/relation/mod.rs

```diff
@@ -1194,12 +1194,58 @@ impl MirRelationExpr {
         aggregates: Vec<AggregateExpr>,
         expected_group_size: Option<usize>,
     ) -> Self {
-        MirRelationExpr::Reduce {
-            input: Box::new(self),
-            group_key: group_key.into_iter().map(MirScalarExpr::Column).collect(),
-            aggregates,
-            monotonic: false,
-            expected_group_size,
+        // Check if we have an unsigned aggregation and add an appropriate cast here
+        // when the output is expected to be unsigned. This is done since rendering
+        // needs to accumulate to a signed number to deal with retractions, and we do
+        // not want to panic there (#17510).
+        let mut unsigned_sums = aggregates
+            .iter()
+            .enumerate()
+            .filter(|(_, agg)| {
+                agg.func == AggregateFunc::SumUInt16 || agg.func == AggregateFunc::SumUInt32
+            })
+            .map(|(idx, _)| idx)
+            .peekable();
+        if unsigned_sums.peek().is_some() {
+            let unsigned_sums = unsigned_sums.collect::<Vec<_>>();
```

Comment:
Nit: I usually annotate the variable with a type instead of using the turbofish, if convenient.
```rust
let unsigned_sums: Vec<_> = unsigned_sums.collect();
```

I think that improves readability, but I realize that this might be subjective.

Reply:
I am pretty spoiled by type annotations in VS Code with rust-analyzer; by changing to the variable annotation, I actually lose some ease of access to the type information in the VS Code rendering, since the variable is no longer directly annotated with its complete type. But I am happy to change to the style you suggest; I saw you were using it in a few other parts of the code, and it's better to keep consistency. Done now!

Reply:
I'd argue that it is a failing of rust-analyzer (or the VS Code interface for it) to give up entirely on showing the type when it is partially annotated! But that's a good call out. I'm not using inline type annotations, so I didn't notice this.